calculator.rb

```diff
@@ -0,0 +1,186 @@
+##### START INITIAL HASHES #####
+
+# store operation, num_1, and num_2 after entry by user
+# important note - sometimes num_1 and num_2 are stored with revised formatting, but sometimes they are not
+user_input = {}
+
+# list all valid operations that can be requested by the user
+valid_operators = {:+ =>["add", "addition", "+"], :- =>["subtract", "subtraction", "-"], :* =>["multiply", "multiplication", "*", "x", "X"], :/ =>["divide", "division", "/"]}
+
+# wip: valid operations for exponential functions and modular functions
+# :** =>["exponent", "exponents", "^", "**"], :% =>["modulo", "modular function", "modularfunction", "%"]
+
+##### END INITIAL HASHES #####
+
+##### START FUNCTIONS #####
+
+# perform addition
+def addition(num_1, num_2)
+  num_1 + num_2
+  return num_1 + num_2
+end
+
+# perform subtraction
+def subtraction(num_1, num_2)
+    num_1 - num_2
+    return num_1 - num_2
+end
+
+# perform multiplication
+def multiplication(num_1, num_2)
+  num_1 * num_2
+  return num_1 * num_2
+end
+
+# perform division
+def division(num_1, num_2)
+  num_1 / num_2
+  return num_1 / num_2
+end
+
+# wip: perform exponential function; not implemented in program
+def exponential_function
+  num_1 ** num_2
+  return num_1 ** num_2
+end
+
+# wip: perform modular function; not implemented in program
+def modular_function
+  num_1 % num_2
+  return num_1 % num_2
+end
+
+# request operation, num_1, and num_2 from user then store in hash
+def request_operations_numbers(user_input, valid_operators)
+  puts "What operation would you like to perform?"
+  operation = gets.chomp
```

Comment:
We talked about this a little in person, but I'll recap here:

I like that you've broken this out as a separate method, but passing all the user input around as a hash is a little unwieldy. Another possible approach is to have one method to read a number, which is called twice, and another to read the operator.